scripts/test-upgrade.sh

```diff
@@ -22,30 +22,54 @@ BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
 OA_DIR="$BASE_DIR/openstack-ansible"
 RPCD_DIR="$BASE_DIR/rpcd"
 
+# Create a backup of /etc/openstack_deploy, if one doesn't already exist
+# NOTE: This gets done by openstack-ansible's scripts/run-upgrade.sh script,
+#       but if we wait until then it'll be too late.
+if [[ ! -d "/etc/openstack_deploy.MITAKA" ]]; then
+  cp -a /etc/openstack_deploy /etc/openstack_deploy.MITAKA
+fi
+
 # TASK #0
 # Bug: https://github.com/rcbops/u-suk-dev/issues/347
 # Issue: Variables files have been reorganized, run a migration script to move
 #        them to the new location.
-"${BASE_DIR}"/scripts/migrate-yaml.py --for-testing-take-new-vars-only \
-    --defaults "${RPCD_DIR}"/etc/openstack_deploy/user_rpco_variables_defaults.yml \
-    --overrides /etc/openstack_deploy/user_extras_variables.yml \
-    --output-file /etc/openstack_deploy/user_rpco_variables_overrides.yml
-"${BASE_DIR}"/scripts/migrate-yaml.py --for-testing-take-new-vars-only \
-  --defaults "${RPCD_DIR}"/etc/openstack_deploy/user_osa_variables_defaults.yml \
-  --overrides /etc/openstack_deploy/user_variables.yml \
-  --output-file /etc/openstack_deploy/user_osa_variables_overrides.yml
+if [[ ! -f "/etc/openstack_deploy/user_rpco_variables_overrides.yml" ]]; then
+  "${BASE_DIR}"/scripts/migrate-yaml.py --for-testing-take-new-vars-only \
+      --defaults "${RPCD_DIR}"/etc/openstack_deploy/user_rpco_variables_defaults.yml \
+      --overrides /etc/openstack_deploy/user_extras_variables.yml \
+      --output-file /etc/openstack_deploy/user_rpco_variables_overrides.yml
+fi
+if [[ ! -f "/etc/openstack_deploy/user_osa_variables_overrides.yml" ]]; then
+  "${BASE_DIR}"/scripts/migrate-yaml.py --for-testing-take-new-vars-only \
+    --defaults "${RPCD_DIR}"/etc/openstack_deploy/user_osa_variables_defaults.yml \
+    --overrides /etc/openstack_deploy/user_variables.yml \
+    --output-file /etc/openstack_deploy/user_osa_variables_overrides.yml
+fi
+# TODO: We need to eventually find a proper way to migrate these secret variables.
+#       For now, we just copy existing secret variables defined previously on the
+#       liberty install. This won't cause any problems since there aren't any new
+#       secret variables in mitaka that are not defined in liberty.
+cp /etc/openstack_deploy/user_extras_secrets.yml \
+   /etc/openstack_deploy/user_rpco_secrets.yml
 cp -a ${RPCD_DIR}/etc/openstack_deploy/*defaults* /etc/openstack_deploy
-rm /etc/openstack_deploy/user_extras_variables.yml /etc/openstack_deploy/user_variables.yml
+rm /etc/openstack_deploy/user_extras_variables.yml \
+   /etc/openstack_deploy/user_variables.yml        \
+   /etc/openstack_deploy/user_extras_secrets.yml
 
 # TASK #1
-# Bug: https://github.com/rcbops/u-suk-dev/issues/293
+# Update Ansible
+cd ${OA_DIR}
+bash scripts/bootstrap-ansible.sh
```

Comment:
@andymcc Good catch on ``pip.conf``. I'll add that task in.
RE: repo-containers, The reason we needed them up before the OSA upgrade script is because that script runs [a playbook](https://github.com/openstack/openstack-ansible/blob/stable/mitaka/scripts/run-upgrade.sh#L152) that targets repo-containers before running ``setup-hosts.yml``. And if they are not there before it's run, it will fail :(